app/app_test.go

```diff
@@ -440,3 +444,85 @@ func TestMsgServerProtoAnnotations(t *testing.T) {
+				Key:      "favorite_flower",
+				Value:    "fuchsia",
+			},
+		},
+	}
+	paramChangePropAny, err := codectypes.NewAnyWithValue(paramChangeProp)
+	require.NoError(t, err, "codectypes.NewAnyWithValue(paramChangeProp)")
+	t.Logf("paramChangePropAny.TypeUrl = %q", paramChangePropAny.TypeUrl)
+
+	execLeg := &govtypesv1.MsgExecLegacyContent{
+		Content:   paramChangePropAny,
+		Authority: "jerry",
+	}
+
+	execLegAny, err := codectypes.NewAnyWithValue(execLeg)
+	require.NoError(t, err, "codectypes.NewAnyWithValue(execLeg)")
+
+	submitTime := time.Unix(1618935600, 0)
+	depositEndTime := submitTime.Add(1 * time.Minute)
+	votingStartTime := depositEndTime.Add(3500 * time.Millisecond)
+	votingEndTime := votingStartTime.Add(48 * time.Hour)
+
+	prop := govtypesv1.Proposal{
+		Id:               123,
+		Messages:         []*codectypes.Any{execLegAny},
+		Status:           govtypesv1.StatusPassed,
+		FinalTallyResult: &govtypesv1.TallyResult{YesCount: "5", AbstainCount: "1", NoCount: "0", NoWithVetoCount: "0"},
+		SubmitTime:       &submitTime,
+		DepositEndTime:   &depositEndTime,
+		TotalDeposit:     []sdk.Coin{sdk.NewInt64Coin("pink", 1000)},
+		VotingStartTime:  &votingStartTime,
+		VotingEndTime:    &votingEndTime,
+		Metadata:         "Prop metadata",
+		Title:            "The prop title",
+		Summary:          "The prop summary",
+		Proposer:         sdk.AccAddress("proposer____________").String(),
+	}
+
+	expJSON := `{"id":"123",` +
+		`"messages":[{"@type":"/cosmos.gov.v1.MsgExecLegacyContent",` +
+		`"content":{"@type":"/cosmos.params.v1beta1.ParameterChangeProposal",` +
+		`"title":"Test Prop Change",` +
+		`"description":"A proposal for testing decoding",` +
+		`"changes":[{"subspace":"mymodule","key":"favorite_flower","value":"fuchsia"}]},` +
+		`"authority":"jerry"}],` +
+		`"status":"PROPOSAL_STATUS_PASSED",` +
+		`"final_tally_result":{"yes_count":"5","abstain_count":"1","no_count":"0","no_with_veto_count":"0"},` +
+		`"submit_time":"2021-04-20T16:20:00Z",` +
+		`"deposit_end_time":"2021-04-20T16:21:00Z",` +
+		`"total_deposit":[{"denom":"pink","amount":"1000"}],` +
+		`"voting_start_time":"2021-04-20T16:21:03.500Z",` +
+		`"voting_end_time":"2021-04-22T16:21:03.500Z",` +
+		`"metadata":"Prop metadata",` +
+		`"title":"The prop title",` +
+		`"summary":"The prop summary",` +
+		`"proposer":"cosmos1wpex7ur0wdjhyh6lta047h6lta047h6ljkx24t",` +
+		`"expedited":false,` +
+		`"failed_reason":""}`
+
+	propBz, err := prop.Marshal()
+	require.NoError(t, err, "prop.Marshal()")
+
+	encCfg := MakeTestEncodingConfig(t)
+
+	var actProp govtypesv1.Proposal
+	err = encCfg.Marshaler.Unmarshal(propBz, &actProp)
+	require.NoError(t, err, "encCfg.Marshaler.Unmarshal(propBz, &actProp)")
+
+	propJSONBz, err := encCfg.Marshaler.MarshalJSON(&actProp)
+	require.NoError(t, err, "encCfg.Marshaler.MarshalJSON(&actProp)")
+	propJSON := string(propJSONBz)
+	t.Logf("prop JSON:\n%s", propJSON)
+	assert.Equal(t, expJSON, propJSON, "proposal JSON")
+}
```

Comment:
This test was written to demonstrate a known failure case that needs to be fixed. It was even written before the fix was implemented. All the test is doing is making sure that our encoder is wired up to correctly handle a gov `Proposal` with a `ParameterChangeProposal` in it. The actual content of the `Proposal` or `ParameterChangeProposal` is inconsequential because the encoders treat it all the same anyway.

These types as well as the encoder types are defined and maintained in imported libraries outside our control. This is not the place to do extensive testing on those types unless there's a known issue that we're trying to work around. That's not the case here. Here, we just need to make sure our things are wired up correctly, which is what this test does.